File: app/src/main/java/com/android/ui/kent/demo/architecture/room/RoomActivity.java
package com.android.ui.kent.demo.architecture.room;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.AppBarLayout;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.android.ui.kent.R;
import com.android.ui.kent.database.room.Member;
import com.android.ui.kent.database.room.MemberDao;
import com.android.ui.kent.database.room.MemberDb;
import com.android.ui.kent.demo.BaseActivity;
import com.android.ui.kent.demo.application.MyApplication;
import com.google.gson.Gson;

import java.util.Date;
import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.reactivex.Observable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by Kent Song on 2019/2/11.
 */
public class RoomActivity extends BaseActivity {


    @BindView(R.id.toolbar_layout)
    AppBarLayout toolbarLayout;
    @BindView(R.id.edit_text)
    EditText editText;
    @BindView(R.id.btn_add)
    Button btnAdd;
    @BindView(R.id.btn_del)
    Button btnDel;
    @BindView(R.id.text_result)
    TextView textResult;

    @Inject
    MemberDb mMemberDb;
    @Inject
    MemberDao mMemberDao;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_room);
        ButterKnife.bind(this);

        MyApplication.getAppComponent().inject(this);
    }

    @OnClick({R.id.btn_add, R.id.btn_del})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btn_add:
                Member member = new Member();
                member.setName("name1");
                member.setAge(18);
                member.setBirthday(new Date());

                Observable.just("")
                        .subscribeOn(Schedulers.io())
                        .doOnNext(new Consumer<String>() {
                            @Override
                            public void accept(String s) throws Exception {
                                mMemberDb.getMemberDao()
                                        .insert(member);
                            }
                        })
                        .doOnNext(new Consumer<String>() {
                            @Override
                            public void accept(String s) throws Exception {
                                queryMember();
                            }
                        })
                        .subscribe();

                break;
            case R.id.btn_del:

                if (TextUtils.isEmpty(editText.getText().toString())) {
                    return;
                }

                Observable.just("")
                        .subscribeOn(Schedulers.io())
                        .doOnNext(new Consumer<String>() {
                            @Override
                            public void accept(String s) throws Exception {
                                Member member = mMemberDao.getMember(Integer.parseInt(editText.getText().toString()));
                                if (member != null) {
                                    mMemberDao.delete(member);
                                }
                            }
                        })
                        .doOnNext(new Consumer<String>() {
                            @Override
                            public void accept(String s) throws Exception {
                                queryMember();
                            }
                        })
                        .subscribe();
                break;
            default:
                break;
        }
    }

    private void queryMember() {
        List<Member> allMembers = MemberDb
                .getInstance(this)
                .getMemberDao()
                .getAllMembers();

        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                StringBuilder sb = new StringBuilder();
                for (Member m : allMembers) {
                    sb.append(new Gson().toJson(m));
                    sb.append(System.lineSeparator());
                }
                textResult.setText(sb.toString());
            }
        });
    }

    public static void launch(Activity activity) {
        Intent intent = new Intent(activity, RoomActivity.class);
        activity.startActivity(intent);
    }
}

File: slimsample/src/main/java/com/kent/android/slim/sample/RoomActivity.kt
package com.kent.android.slim.sample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.room.Room
import com.google.gson.Gson
import com.kent.android.slim.sample.room.AppDatabase
import com.kent.android.slim.sample.room.gift.GiftUsage
import com.kent.android.slim.sample.room.gift.GiftUsageDao
import com.kent.android.slim.sample.room.user.User
import com.kent.android.slim.sample.room.user.UserDao
import com.kent.android.slim.sample.util.CoroutineExecutor
import kotlinx.android.synthetic.main.activity_room.*
import java.lang.StringBuilder

/**
 * Created by Kent Sung on 2021/10/14.
 */
class RoomActivity : AppCompatActivity() {

    lateinit var userDao: UserDao
    lateinit var giftDao: GiftUsageDao
    var i = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_room)

        val db = Room.databaseBuilder(
            applicationContext,
            AppDatabase::class.java, "database-name"
        ).build()

        userDao = db.userDao();
        giftDao = db.giftUsageDao()

        btn_add.setOnClickListener {
            CoroutineExecutor.launchIO {
                val user = User(i++, "kent", "song")
                userDao.insertAll(user)
                queryAll()
            }

        }

        btn_gift_add.setOnClickListener {
            CoroutineExecutor.launchIO {
                i++
                val giftUseage = GiftUsage("gift_id_1", "2020-11-08"+i)
                giftDao.insert(giftUseage)
                queryAll()
            }
        }

        btn_gift_delete.setOnClickListener{
            CoroutineExecutor.launchIO {
//                giftDao.deleteById("gift_id_1")
                giftDao.deleteAll()
                queryAll()
            }
        }
    }

    fun queryAll() {
        val allUser = userDao.getAll()
        val allGift = giftDao.getAll()

        runOnUiThread {
            val sb = StringBuilder()
            for (m in allUser) {
                sb.append(Gson().toJson(m))
                sb.append(System.lineSeparator())
            }
            for(item in allGift){
                sb.append(Gson().toJson(item))
                sb.append(System.lineSeparator())
            }

            text_result.text = sb.toString()
        }
    }
}